Analysis/include/Coda2EventDecoder.h

```diff
@@ -0,0 +1,32 @@
+#ifndef CODA2EVENTDECODER_H
+#define CODA2EVEMTDECODER_H
+
+#include "VEventDecoder.h"
+#include "Rtypes.h"
+
+#include <vector>
+
+class Coda2EventDecoder : public VEventDecoder
+{
+public:
+		Coda2EventDecoder() { }
+		~Coda2EventDecoder() { }
+public:
+// Encoding Functions
+	virtual std::vector<UInt_t> EncodePHYSEventHeader();
+	virtual void EncodePrestartEventHeader(int* buffer, int buffer_size, int runnumber, int runtype = 0);
+  virtual void EncodeGoEventHeader(int* buffer, int buffer_size);
+  virtual void EncodePauseEventHeader(int* buffer, int buffer_size);
+  virtual void EncodeEndEventHeader(int* buffer, int buffer_size);
+public:
+// Decoding Functions
+  virtual Int_t DecodeEventIDBank(UInt_t *buffer);
+	virtual void PrintDecoderInfo(QwLog& out);
```

Comment:
There are some instances where we display key decoder information for reference or debugging purposes inside QwEventBuffer. This leads to some inconveniences:
a) the decoder data members are currently public (BAD!)
and
b) having decoder->var throughout the QwEventBuffer code is ugly (BAD!)

Abstracting the output of key decoder information solves both these issues. You can pass QwDebug, QwMessage, or QwWarn into the signature to dictate the verbosity.